orangecontrib/text/widgets/owcollocations.py

```diff
@@ -0,0 +1,191 @@
+                                          callback=self.commit)
+
+        gui.rubber(self.controlArea)
+
+        gui.button(self.buttonsArea, self, "Restore Original Order",
+                   callback=self.restore_order,
+                   tooltip="Show rows in the original order",
+                   autoDefault=False)
+
+        # GUI
+        self.collModel = model = TableModel(parent=self)  # type: TableModel
+        model.setHorizontalHeaderLabels(["Method", "Score"])
+        self.collView = view = TableView(self)  # type: TableView
+        self.mainArea.layout().addWidget(view)
+        view.setModel(model)
+        view.resizeColumnsToContents()
+        view.setItemDelegateForColumn(1, gui.ColoredBarItemDelegate())
+        view.setSelectionMode(QTableView.NoSelection)
+
+    @Inputs.corpus
+    def set_corpus(self, corpus):
+        self.collModel.clear()
+        self.collModel.resetSorting(True)
+        self.corpus = corpus
+        self.commit()
+
+    def _change_type(self):
+        self.type = NGRAM_TYPES[self.type_index]
+        if self.type_index == 1:
+            self.method_rb.buttons[2].setDisabled(True)
+            self.method_rb.buttons[3].setDisabled(True)
+            self.method_rb.buttons[7].setDisabled(True)
+            if self.selected_method in [2, 3, 7]:
+                self.method_rb.buttons[0].click()
+        else:
+            self.method_rb.buttons[2].setDisabled(False)
+            self.method_rb.buttons[3].setDisabled(False)
+            self.method_rb.buttons[7].setDisabled(False)
+        self.commit()
+
+    def compute_scores(self):
+        self.collModel.clear()
+        self.collModel.resetSorting(True)
+        finder = self.type.from_documents(self.corpus.tokens)
+        finder.apply_freq_filter(self.freq_filter)
+
+        res = finder.score_ngrams(self.method.scorer[self.type_index])
+        collocations = np.array([" ".join(col) for col, score in res],
+                                dtype=object)[:, None]
+        scores = np.array([score for col, score in res], dtype=float)[:, None]
+
+        self.results = (collocations, scores)
+        if len(scores):
+            self.collModel.setExtremesFrom(1, scores)
+
+    def commit(self):
+        if self.corpus is None:
+            return
+
+        self.type = NGRAM_TYPES[self.type_index]
+        self.method = SCORING_METHODS[self.selected_method]
+
+        self.compute_scores()
+
+        if not self.results:
+            self.collModel.clear()
+            self.Outputs.corpus.send(None)
+            return
+
+        output = self.create_scores_table()
+        self.collModel[:] = np.hstack(self.results)[:20]
+        self.collView.resizeColumnsToContents()
+
+        self.Outputs.corpus.send(output)
+
+    def create_scores_table(self):
+        col_var, score_var = get_unique_names(self.corpus.domain,
+                                              ["Collocations", "Scores"])
+
```

Comment:
```suggestion
```

I think it is unnecessary since you are creating a new table with only Collocations and Scores columns, so you know those two names are unique. Input table columns are not used in the new table.